### PR TITLE
Jira integration

### PR DESCRIPTION
**Pull request by mati007thm**
fixes https://github.com/mondoohq/terraform-provider-mondoo/issues/160

### CODE REVIEW

File: docs/resources/integration_jira.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  space = "hungry-poet-123456"
+}
+
+# Setup the Jira integration
+resource "mondoo_integration_jira" "jira_integration" {
+  name  = "My Jira Integration"
+  host  = "https://your-instance.atlassian.net"
+  email = "jira.owner@email.com"
+  # default_project = "MONDOO"
+  api_token = var.jira_token
```
Comment (chris-rock) [inline docs/resources/integration_jira.md:32]: Token needs to become a credentials block, to align with the other integrations:

```
 credentials = {
    api_token = var.jira_token
  }
  ```

Comment (afiune) [reply docs/resources/integration_jira.md:32]: Same ☝🏽

File: internal/provider/integration_jira_resource.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+				MarkdownDescription: "Jira user email",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$`),
+						"must be a valid email",
+					),
+				},
+			},
+			"api_token": schema.StringAttribute{
```
Comment (afiune) [inline internal/provider/integration_jira_resource.go:95]: more to a credentials object, like the other integrations

File: internal/provider/integration_jira_resource.go [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+	ctx = tflog.SetField(ctx, "space_mrn", space.MRN())
+
+	// Do GraphQL request to API to create the resource.
+	tflog.Debug(ctx, "Creating integration")
+	integration, err := r.client.CreateIntegration(ctx,
+		space.MRN(),
+		data.Name.ValueString(),
+		mondoov1.ClientIntegrationTypeTicketSystemJira,
+		mondoov1.ClientIntegrationConfigurationInput{
+			JiraConfigurationOptions: &mondoov1.JiraConfigurationOptionsInput{
```
Comment (afiune) [inline internal/provider/integration_jira_resource.go:161]: Follow other patterns to `GetConfigurationOptions()`

File: internal/provider/integration_jira_resource.go [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+	// Do GraphQL request to API to update the resource.
+	_, err := r.client.DeleteIntegration(ctx, data.Mrn.ValueString())
+	if err != nil {
+		resp.Diagnostics.
+			AddError("Client Error",
+				fmt.Sprintf("Unable to delete Jira integration, got error: %s", err),
+			)
+		return
+	}
+}
```
Comment (afiune) [inline internal/provider/integration_jira_resource.go:262]: Missing import function

**Review by afiune — CHANGES_REQUESTED** [review]
Add test similar to https://github.com/mondoohq/terraform-provider-mondoo/blob/main/internal/provider/integration_shodan_resource_test.go

File: docs/resources/integration_jira.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "mondoo_integration_jira Resource - terraform-provider-mondoo"
```
Comment (afiune) [inline docs/resources/integration_jira.md:3]: Fix these spelling issues via https://github.com/mondoohq/terraform-provider-mondoo/tree/main/.github/actions/spelling

**Comment by mati007thm** [comment]
@afiune I included the tests and made the requested changes

**Review by afiune — APPROVED** [review]

**Comment by afiune** [comment]
@mati007thm Need to resolve the conflicts, then we will merge! 🙌🏽

**Comment by afiune** [comment]
Also look at https://github.com/mondoohq/terraform-provider-mondoo/pull/180, you might need to resolve something in the acceptance tests.

**Comment by mati007thm** [comment]
@afiune I changed the acceptance tests. Hope this works now!

**Review by afiune — APPROVED** [review]

**Pull request merged**